### ISSUE TITLE
Can I reset SRB miner?

### ISSUE THREAD

**Issue by Djago** [MultiPoolMiner/MultiPoolMiner#2259]
As I posted in another issue, my VEGA card tends to drop hashrate from 1850 to 1100 (SRB). Probable causes are: starting the miner a couple of minutes after starting Windows, starting a video file (local or streaming, but not always!), sometimes it simply drops hashrate. Then it stays constant at that value. I have to restart or suspend/resume Windows to recover the speed. If I run at full speed and I don't do anything fancy, miner can stay at full speed for days without a glitch.
I've found that the problem is the VEGA related and it's a PITA, but some miners have internal watchdogs to remedy this. I've found that SRB miner has 2 parameters: "min_rig_speed" : xxxx (in H/s) in the config, and --resetvega (in the command line). First parameter detects when hashrate has an average (5') value lower than xxxx and then restart the miner. The second parameter tells the miner to restart the VEGA (it's analogous to disable/enable the card in the device manager), but you have to run the file with admin privileges.
Now, the question is: can I add both parameters to the command line in MPM without breaking anything? I mean, if MPM sees that SRB is restarting, won't it interpret that as a failure?

**Comment by UselessGuru** [comment]
> starting a video file (local or streaming, but not always!), sometimes it simply drops hashrate. Then it stays constant at that value. I

Try looking at a combination of OC software like MSI Afterburner. scheduled tasks and events in the event log. I assume you will find a corresponding event in the event log for every time the card switches to lower speed. Use a custom scheduled task to take action on the event to re-apply your OC settings with Afterburner, e.g. like this:

![image](https://user-images.githubusercontent.com/30080938/49133077-f9a80600-f2de-11e8-8149-7c5185c9a55f.png)

![image](https://user-images.githubusercontent.com/30080938/49133096-075d8b80-f2df-11e8-8aa8-8f25a26b227c.png)

> 'min_rig_speed'

I'll see if I can implement this.

> '-resetvega'

MPM would probably see this as a failed miner (red messages).

**Comment by Djago** [comment]
Many thanks for your help, I'll take a look at what you said.

I'm well aware that this is not an MPM problem, but I thought that as there are lots of VEGA miners, maybe one haveing the same issue read my messages! I've looked for help in other forums and in one of those forums (http://vega.miningguides.com) is where I've learned that this is something common for the VEGA card and the solution is to create a bat and execute it with admin priv that makes this reset, or use a miner that do this reset on it's own (like SRB). That's why I've asked here if I can use that switch!
